src/cmd/build.js

```diff
@@ -124,11 +130,31 @@ async function defaultPackageCreator({
   const packageName = safeFileName(
     `${extensionName}-${manifestData.version}.zip`);
   const extensionPath = path.join(artifactsDir, packageName);
-  const stream = createWriteStream(extensionPath);
+
+  function throwExtensionExists() {
+    throw new UsageError(
+      `Extension exists at the destination path: ${extensionPath}\n` +
+      'Use --overwrite-dest to enable overwriting.');
+  }
+
+
+  if (overwriteDest) {
+    log.info(`Destination exists, overwriting: ${extensionPath}`);
+  }
+
+  const stream = overwriteDest ?
+            createWriteStream(extensionPath)
+           : createWriteStream(extensionPath, {flags: 'wx'}); // double-check if destination file exists
 
   stream.write(buffer, () => stream.end());
 
-  await eventToPromise(stream, 'close');
+  try {
+    await eventToPromise(stream, 'close');
+  } catch (error) {
+    if (isErrorWithCode('EEXIST', error)) {
+      throwExtensionExists();
+    }
```

Comment:
we should re-raise any unexpected error here (basically anything that is not an 'EEXIST' error.